FusionIIIT/applications/research_procedures/views.py

```diff
@@ -133,7 +133,15 @@ def consult_insert(request):
     messages.success(request,"Successfully created consultancy project")
     return redirect(reverse("research_procedures:patent_registration"))
 
+@login_required
 def add_projects(request):
+
+    # designation = getDesignation(request.user.username)
+    # print("designation is " + designation)
+    # if designation != 'rspc_admin':
+    #     messages.error(request, 'Only RSPC Admin can add projects')
+    #     return redirect("/research_procedures")
+
     if request.method== "POST":
         obj= request.POST
         projectname= obj.get('project_name')
```

Comment:
Please break this function into modular small function since it has a very important functionality and is quite big